Sources/Widgets/Core/WidgetManager/index.js

```diff
@@ -253,30 +253,41 @@ function vtkWidgetManager(publicAPI, model) {
   // API public
   // --------------------------------------------------------------------------
 
-  async function updateSelection({ position }) {
-    model._selectionInProgress = true;
+  async function updateSelection(callData, fromTouchEvent, callID) {
+    const { position } = callData;
     const { requestCount, selectedState, representation, widget } =
       await publicAPI.getSelectedDataForXY(position.x, position.y);
-    model._selectionInProgress = false;
 
-    if (requestCount) {
-      // Call activate only once
+    if (requestCount || callID !== model._currentUpdateSelectionCallID) {
+      // requestCount > 0: Call activate only once
+      // callID check: drop old calls
       return;
     }
 
+    function activateHandle(w) {
+      if (fromTouchEvent) {
+        // release any previous left button interaction
+        model._interactor.invokeLeftButtonRelease(callData);
+        w.activateHandle({ selectedState, representation });
+        model._interactor.invokeLeftButtonPress(callData);
```

Comment:
I typically prefer code factorization to avoid diverging in the future:
```
if (fromTouchEvent) {
  model._interactor.invokeLeftButtonRelease(callData);
}
w.activateHandle({ selectedState, representation });
if (fromTouchEvent) {
        model._interactor.invokeLeftButtonPress(callData);
}
```

Reply:
Done!